### PR TITLE
Use X-macros to simplify new setting creation in SettingsModel

### PR DESCRIPTION
**Pull request by PankajBhojwani**
Introduces X-macros to reduce the number of places we need to write essentially the same line of code but for a different setting (declaring it in the header file, in `Copy`, `LayerJson`, `ToJson`, etc).

### CODE REVIEW

File: src/cascadia/TerminalSettingsModel/TerminalSettings.h [hunk 1]
```diff
@@ -84,21 +85,29 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         INHERITABLE_SETTING(Model::TerminalSettings, til::color, DefaultForeground, DEFAULT_FOREGROUND);
         INHERITABLE_SETTING(Model::TerminalSettings, til::color, DefaultBackground, DEFAULT_BACKGROUND);
         INHERITABLE_SETTING(Model::TerminalSettings, til::color, SelectionBackground, DEFAULT_FOREGROUND);
-        INHERITABLE_SETTING(Model::TerminalSettings, int32_t, HistorySize, DEFAULT_HISTORY_SIZE);
-        INHERITABLE_SETTING(Model::TerminalSettings, int32_t, InitialRows, 30);
-        INHERITABLE_SETTING(Model::TerminalSettings, int32_t, InitialCols, 80);
 
-        INHERITABLE_SETTING(Model::TerminalSettings, bool, SnapOnInput, true);
-        INHERITABLE_SETTING(Model::TerminalSettings, bool, AltGrAliasing, true);
+#define GLOBAL_SETTINGS_INITIALIZE(type, name, ...) \
```
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsModel/TerminalSettings.h:89]: Hmm. I wonder if we get more utility by doing the `TerminalSettings` bits with the Control/Core settings macro that I'm doing over in [`dev/migrie/oop/ragnarok`](https://github.com/microsoft/terminal/compare/dev/migrie/oop/ragnarok). That way we could stuff more things into the Global/Profile/Appearance macros as-is. 

Idk that's idle musing without too much thinking about it

**Review by Microsoft-code — APPROVED** [review]

**Comment by NotWearingPants** [comment]
I love this, will you do the same for actions?

Also, can you use this to generate the settings schema, or the other way around? Using the C-preprocessor to generate JSON is weird but possible, but maybe a python script will do

**Review by zadjii-msft — DISMISSED** [review]

**Comment by DHowett** [comment]
Unfortunately, new policies require re-reviews when new commits are pushed.

File: src/cascadia/TerminalSettingsModel/Profile.cpp [hunk 2]
```diff
@@ -117,28 +102,19 @@ winrt::com_ptr<Profile> Profile::CopySettings() const
...
-    profile->_TabTitle = _TabTitle;
     profile->_TabColor = _TabColor;
-    profile->_SuppressApplicationTitle = _SuppressApplicationTitle;
-    profile->_UseAcrylic = _UseAcrylic;
-    profile->_ScrollState = _ScrollState;
     profile->_Padding = _Padding;
-    profile->_Commandline = _Commandline;
-    profile->_StartingDirectory = _StartingDirectory;
-    profile->_AntialiasingMode = _AntialiasingMode;
     profile->_ForceFullRepaintRendering = _ForceFullRepaintRendering;
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/Profile.cpp:107]: i'm surprised you didn't move these into the macro . . . but it seems like you _did_?

Comment (DHowett) [reply src/cascadia/TerminalSettingsModel/Profile.cpp:107]: I'm sorry, what? This is supported as a profile setting?

Comment (DHowett) [reply src/cascadia/TerminalSettingsModel/Profile.cpp:107]: I'm very confused.

**Review by DHowett — CHANGES_REQUESTED** [review]
oen q about the rendering settings -- are they per profile or are they per globals?

File: src/cascadia/TerminalSettingsModel/Profile.cpp (hunk 2 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsModel/Profile.cpp:107]: They are in the global settings macro, not in the profile settings macro

Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsModel/Profile.cpp:107]: Because we don't actually use them in `Profile::LayerJson` or `Profile::ToJson`

**Review by zadjii-msft — APPROVED** [review]

File: src/cascadia/TerminalSettingsModel/Profile.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/Profile.cpp:107]: So... they just should have never been in `Profile` in the first place?

Comment (DHowett) [reply src/cascadia/TerminalSettingsModel/Profile.cpp:107]: @PankajBhojwani, can you follow up (separate PR) to remove these from Profile if they're not used? Thx

**Review by DHowett — APPROVED** [review]

**Pull request merged**